conda/core/prefix_data.py

```diff
@@ -589,6 +591,18 @@ def _write_environment_state_file(self, state: dict[str, dict[str, str]]) -> Non
         env_vars_file = self.prefix_path / PREFIX_STATE_FILE
         env_vars_file.write_text(json.dumps(state, ensure_ascii=False))
 
+    def _ensure_no_reserved_env_vars(self, env_vars_names: Iterable[str]) -> None:
+        """
+        Ensure that the set of env_var_names does not contain any reserved env vars.
+        Will raise an OperationNotAllowed if a reserved env var is found.
+        """
+        for var in RESERVED_ENV_VARS:
```

Comment:
Should we collect vars that are in Reserved ENV Vars so all the vars are declared at once?